modules/mod-deep-learning/DeepModel.h

```diff
@@ -0,0 +1,117 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepModel.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class DeepModel
+\brief base class for handling torchscript models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include "ModelCard.h"
+#include "wx/log.h"
+
+#include <tuple>
+
+#include <torch/script.h>
+#include <torch/torch.h>
+#include "AudacityException.h"
+#include "Identifier.h"
+
+class DeepModel;
+
+using ModulePtr = std::unique_ptr<torch::jit::script::Module>;
+using DeepModelHolder = std::shared_ptr<DeepModel>;
+using TensorWithTimestamps = std::tuple<torch::Tensor, torch::Tensor>;
+
+class ModelException final : public MessageBoxException
+{
+public:
+   ModelException(const TranslatableString msg, std::string trace) :
+                  m_msg(msg),
+                  m_trace(std::move(trace)),
+                  MessageBoxException{
+                     ExceptionType::Internal,
+                     XO("Deep Model Error")
+                  }
+   { 
+      if (!m_trace.empty()) 
+         wxLogError(wxString(m_trace)); 
+   }
+
+   // internal message
+   virtual const char* what() const throw () 
+      { return m_msg.Translation().c_str(); }
+
+   // user facing message
+   virtual TranslatableString ErrorMessage() const
+      { return XO("Deep Model Error: %s").Format(m_msg);}
+   
+   const TranslatableString m_msg;
+   const std::string m_trace;
+};
+
+class DeepModel final
+{
+public:
+   // base directory for deep learning models
+   // TODO: maybe we want to support a couple of search paths?
+   static FilePath DLModelsDir();
+   static FilePath BuiltInModulesDir();
+
+   DeepModel() = default;
+   DeepModel(ModelCard &card);
+
+   // @execsafety: strong 
+   // load a torchscript model along with it's metadata, 
+   // which is stored in a ModelCard.
+   void Load(const std::string &modelPath);
+   void Load(std::istream &bytes);
```

Comment:
Might preced with
`//! @copydoc Load(std::string&)`

Reply:
resolved in 8cce7502a